patches/server/0930-Add-EntityFertilizeEggEvent.patch

```diff
@@ -5,37 +5,100 @@ Subject: [PATCH] Add EntityFertilizeEggEvent
++        final io.papermc.paper.event.entity.EntityFertilizeEggEvent result = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityFertilizeEggEvent(this, other);
++        if (result.isCancelled()) return;
++        // Paper end - Add EntityFertilizeEggEvent event
++
+         ItemStack itemstack = new ItemStack(Items.SNIFFER_EGG);
+         ItemEntity entityitem = new ItemEntity(world, this.position().x(), this.position().y(), this.position().z(), itemstack);
+ 
+         entityitem.setDefaultPickUpDelay();
+-        this.finalizeSpawnChildFromBreeding(world, other, (AgeableMob) null);
++        this.finalizeSpawnChildFromBreeding(world, other, (AgeableMob) null, result.getExperience()); // Paper - Add EntityFertilizeEggEvent event
+         this.playSound(SoundEvents.SNIFFER_EGG_PLOP, 1.0F, (this.random.nextFloat() - this.random.nextFloat()) * 0.2F + 0.5F);
+         world.addFreshEntity(entityitem);
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index 86b0fe7b6feeef0e085e577954f869e020cc0f04..89235dc1539a7e280d85bf3d998b0a32ed20216f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -1981,4 +1981,29 @@ public class CraftEventFactory {
+         return event.callEvent();
+     }
+     // Paper end
++
++    // Paper start - add EntityFertilizeEggEvent
++    /**
++     * Calls the io.papermc.paper.event.entity.EntityFertilizeEggEvent.
++     * If the event is cancelled, this method also resets the love on both the {@code breeding} and {@code other} entity.
++     *
++     * @param breeding the entity on which #spawnChildFromBreeding was called.
++     * @param other    the partner of the entity.
++     * @return the event after it was called. The instance may be used to retrieve the experience of the event.
++     */
++    public static io.papermc.paper.event.entity.EntityFertilizeEggEvent callEntityFertilizeEggEvent(net.minecraft.world.entity.animal.Animal breeding,
++                                                                                                    net.minecraft.world.entity.animal.Animal other) {
++        net.minecraft.server.level.ServerPlayer serverPlayer = breeding.getLoveCause();
++        if (serverPlayer == null) other.getLoveCause();
++        final int experience = breeding.getRandom().nextInt(7) + 1; // From Animal#spawnChildFromBreeding(ServerLevel, Animal)
++
++        final var event = new io.papermc.paper.event.entity.EntityFertilizeEggEvent((org.bukkit.entity.LivingEntity) breeding.getBukkitEntity(), (org.bukkit.entity.LivingEntity) other.getBukkitEntity(), serverPlayer == null ? null : serverPlayer.getBukkitEntity(), breeding.breedItem == null ? null : org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(breeding.breedItem).clone(), experience);
```

Comment:
You can use CraftItemStack#asCraftCopy

Reply:
asCraftCopy takes a bukkit item stack. I am confused what you mean here.

Reply:
oh nvm